### ISSUE TITLE
Make its input and output Linux pipe friendly

### ISSUE THREAD

**Issue by ibnesayeed** [oduwsdl/sumgram#17]
The tool should accept input from the `STDIN` and generate output to the `STDOUT` so that it can work nicely with other Unix tools.

**Comment by phonedude** [comment]
+1 

I also noted this with "scraper" in yesterday's telecon too.

**Comment by anwala** [comment]
@ibnesayeed,

For a single text collection, this shouldn't be a problem.

However, for multiple text collections, we'd need an unambiguous boundary marker to determine the end of "text collection 1" and the beginning of another. When reading separate files, this is not a problem because the files serve as natural text collection boundaries.

For a single text collection, is this usage what you have in mind?:
`$ sumgram "The eye of Category 4 Hurricane Harvey..."`

**Comment by ibnesayeed** [comment]
> For a single text collection, is this usage what you have in mind?:
> `$ sumgram "The eye of Category 4 Hurricane Harvey..."`

No, this style of argument was not the intent, because that would be ambiguous and no one wants to supply a long text as an argument. Please let the argument reserved for file/folder paths and use STDIN as input instead, so something like the following works:

```
$ echo "The eye of Category 4 Hurricane Harvey..." | sumgram
```

**Comment by ibnesayeed** [comment]
FYI, many things that you trying to do with the input (e.g., reading from STDIN, one or more files as arguments, compressed files, etc.) can be handled more easily using the [fileinput](https://docs.python.org/3/library/fileinput.html) module in the standard Python library.

**Comment by anwala** [comment]
Oh, I see, I misunderstood. Thanks a lot for the clarification.
I'll ensure you can read text from STDIN & consider using fileinput.
